src/components/Input/QtySelector/QtySelector.js

```diff
@@ -30,52 +31,80 @@ class QtySelector extends Component {
   static propTypes = {
     disabled: PropTypes.bool,
     value: PropTypes.number,
-    style: PropTypes.objectOf(PropTypes.string)
+    style: PropTypes.objectOf(PropTypes.string),
+    min: PropTypes.number,
+    max: PropTypes.number,
+    onValueUpdated: PropTypes.func
   };
 
   static defaultProps = {
     disabled: false,
     style: {},
-    value: null
+    value: 0,
+    min: QtySelector.MIN_INPUT_VALUE,
+    max: QtySelector.MAX_INPUT_VALUE,
+    onValueUpdated: () => {}
   };
 
   state = {
-    value: this.props.value || "",
+    value: this.props.value,
     focused: false
   };
 
   handleChange = e => {
     const { value } = e.target;
+    const { max, min } = this.props;
 
     if (
       QtySelector.isNumber(value) &&
-      value.length <= QtySelector.MAX_LENGTH_VAL
+      value.length <= QtySelector.MAX_LENGTH_VAL &&
+      value <= max &&
+      value >= min
     ) {
-      this.setState(() => ({ value: Number.parseInt(value, 10) }));
+      this.setState(
+        () => ({ value: Number.parseInt(value, 10) }),
+        () => this.handleValueUpdated()
+      );
     } else if (value === "") {
       // delete input value case
-      this.setState(() => ({ value: "" }));
+      this.setState(() => ({ value: "" }), () => this.handleValueUpdated());
     }
   };
 
-  increment = () =>
-    this.setState(state => {
-      if (state.value === "") {
-        return { value: 0 };
-      }
-      if (state.value === QtySelector.MAX_INPUT_VALUE) {
-        return { value: QtySelector.MAX_INPUT_VALUE };
-      }
-      return { value: state.value + 1 };
-    });
-
-  decrement = () =>
-    this.setState(state => {
-      if (state.value === "" || state.value === 0) {
-        return { value: 0 };
-      }
-      return { value: state.value - 1 };
-    });
+  increment = () => {
+    const { max } = this.props;
+    this.setState(
+      state => {
+        if (state.value === "") {
```

Comment:
I don't believe this `state.value === ""` case is necessary now that the `value` prop is defaulted to 0 🤔

Reply:
The `state.value === ""` still need to exist because we need to support an empty string when the input field is manually entered using the keyboard but then left to be empty. I just pushed a commit to better handle this scenario and reflect the feedback for early return: https://github.com/ticketmaster/aurora/pull/581/commits/392410de0332984e5fb0f18bb5a8af448e226e32

Here's a gif for this special case:
![QtySelector](https://user-images.githubusercontent.com/6036477/62399570-1b722080-b531-11e9-9b08-fe8f6ae356ee.gif)

Please let me know if you have any feedback

Reply:
Ah, got it! I agree! Thanks for clarifying